src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java

```diff
@@ -83,7 +89,24 @@ void configureMavenExecution(MavenExecutionRequest request, List<RepoFromEnv> re
             logRepositoriesAndMirrors(request);
 
             Profile repositoriesFromEnv = new Profile();
-
+            
+            if(addDefaultRepos) {
```

Comment:
This will IMHO duplicate the repositories, as they are already listed in another profile in settings.xml. According to my observations in https://issues.apache.org/jira/browse/MNG-6994 it should be sufficient to just insert the new settings profile at the beginning of the list instead of adding it to the end.

Reply:
Seems there is no API for adjusting the order of profiles in https://maven.apache.org/ref/3.3.1/maven-core/apidocs/org/apache/maven/execution/MavenExecutionRequest.html

Reply:
Actually `request.getProfiles().add(0, repositoriesFromEnv)` should do the trick here.

Reply:
That's a neater solution, followed that approach in a3f6c61c849c27232c. I still left a property to be able to control the behaviour (for the mirror repo case), but the default is switched around compared to currently released version